composer/trainer/checkpoint.py

```diff
@@ -25,14 +26,29 @@ class CheckpointLoader:
         checkpoint_filepath (str): The path to an existing checkpoint file.
     """
 
-    def __init__(self, checkpoint_filepath: str):
+    def __init__(self, checkpoint_filepath: str, load_weights_only: bool = False, strict: bool = False):
         self.state_dict = torch.load(checkpoint_filepath, map_location='cpu')
+        self.load_weights_only = load_weights_only
+        self.strict = strict
 
     def load_checkpoint(self, state: State):
         """Initialize state from the loaded checkpoint's data.
         """
 
-        state.load_state_dict(self.state_dict["state"])
+        if self.load_weights_only:
+
+            if self.state_dict['state']["_is_model_ddp_wrapped"] and not isinstance(state.model,
```

Comment:
Can you refactor some of this logic out of state.load_state_dict so it's not duplicated? Perhaps create a method on the state class called `def load_model_state(self, state_dict: Dict[str, Any], strict: bool)`. Then the `load_state_dict` and state and this logic can call that.

Reply:
Done!